### DISCUSSION TITLE
Twitter Oauth 2.0 problems (Whoa there!)

### DISCUSSION THREAD

**Discussion by BillyLand** [nextauthjs/next-auth#3618] [Help]
Hi, 

I'm trying to get Twitter OAuth 2.0 to work but cannot figure out what i am doing wrong.

I have the latest version of next-auth (4.1.0).
I have set up my twitter developer portal (I have a project, 2.0 is enabled, i have the 2.0 client id and secret and so on)

My next-auth config looks like this:
`TwitterProvider({
				clientId: TWITTER_CLIENT_ID,
				clientSecret: TWITTER_CLIENT_SECRET,
				version: '2.0', // opt-in to Twitter OAuth 2.0
			}),`

I import the default import from next-auth/providers/twitter.

now, what I get is this message:
Whoa there!
There is no request token for this page. That's the special key we need from applications asking to use your Twitter account. Please go back to the site or application that sent you here and try again; it was probably just a mistake.

What am I missing? From my point of view i've done everyting according to docs.

**Comment by balazsorban44** [comment]
have a reproduction?

**Comment by BenPust** [comment]
edit*
@BillyLand I was able to reproduce the issue when running next-auth version 3. (I wanted to integrate firebase adapter which doesn't run on version 4). If I upgrade next-auth to version 4 this solves the issue.

**Comment by pro123908** [comment]
Any update on this, I'm getting the same error
